### PR TITLE
Shoot from players

### PR DESCRIPTION
**Pull request by Magicpig55**
You can shoot from players, and the shots are from the origin. We could do some other things, like having players spawn on sides, but I might save that for a different branch.

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -256,15 +262,23 @@ function AttemptGraph(code, ctx, collisiondata, x, first) {
...
+        var y1 = (height - code.eval(obj)) - (height - y)
         obj.x++
-        var y2 = height - code.eval(obj)
-        var res = CheckLineCollision(collisiondata, x-1, math.floor(y1), x, math.floor(y2))
+        var y2 = (height - code.eval(obj)) - (height - y)
+        var res = 0
+        if(reverse) {
+            // var tempy = y1
+            // y1 = y2
+            // y2 = tempy
```
Comment (jeffnuss) [inline script.js]: Let's clean up this commented out code.

File: script.js [hunk 2]
```diff
@@ -278,7 +292,11 @@ function AttemptGraph(code, ctx, collisiondata, x, first) {
             y2 = res.y
             hit = true
         }
-        var ent = CheckEntityLineCollision(x-1, math.floor(y1), x, math.floor(y2))
+        var ent = 0
+        if (reverse) 
+            ent = CheckEntityLineCollision(xoffset - x, math.floor(y1), xoffset - x + 1, math.floor(y2))
+        else
+            ent = CheckEntityLineCollision(x - 1 + xoffset, math.floor(y1), x + xoffset, math.floor(y2))
```
Comment (jeffnuss) [inline script.js]: Make the parameters that are different variables that get set in the conditional and then just call the method with those vars. Something like:

```
var x1 = 0
var x2 = 0
if (reverse) 
    x1 = xoffset - x
    x2 = xoffset - x + 1
else
    x1 = x - 1 + xoffset
    x2 = x + xoffset

ent = CheckEntityLineCollision(x1, math.floor(y1), x2, math.floor(y2))
```

File: script.js [hunk 3]
```diff
@@ -286,23 +304,35 @@ function AttemptGraph(code, ctx, collisiondata, x, first) {
...
             DrawEntities()
         }
-        ctx.moveTo(x - 1, y1)
-        ctx.lineTo(x, y2)
+        if(reverse) {
+            ctx.moveTo(xoffset - x + 1, y1)
+            ctx.lineTo(xoffset - x, y2)
+        } else {
+            ctx.moveTo(x - 1 + xoffset, y1)
+            ctx.lineTo(x + xoffset, y2)
```
Comment (jeffnuss) [inline script.js]: Do something similar as above.

**Comment by jeffnuss** [comment]
First pass done. See my comments.

**Comment by jeffnuss** [comment]
:+1: looks good to me

File: script.js [hunk 4]
```diff
@@ -136,8 +137,8 @@ function Setup() {
     // Lazily set up entities for testing stuff, probably want to
     // Attach these to network things or something else 
     for(var i = 0; i < 3; i++) {
-        Entities.push(new Entity(math.random(720), math.random(480), Teams.Blue))
-        Entities.push(new Entity(math.random(720), math.random(480), Teams.Orange))
+        Entities.push(new Entity(math.floor(math.random(720)), math.floor(math.random(480)), Teams.Blue))
+        Entities.push(new Entity(math.floor(math.random(720)), math.floor(math.random(480)), Teams.Orange))
```
Comment (jcdiddi) [inline script.js]: If you place the entities at the decimal points, will the start of the graph be more accurate?

Comment (jcdiddi) [inline script.js]: Do we want ID's for each entity?

**Comment by jcdiddi** [comment]
Also, your external JS resources all have TONS of comments. If we deleted those comments, they would be able to download and run faster.

**Comment by Magicpig55** [comment]
At the moment I'm not even referencing those external libraries, I'm only using the minified, gzipped versions provided from the cdns.

**Comment by jeffnuss** [comment]
CDN versions FTW. I'd say we get rid of our versions of them then.

**Pull request merged**